docs/src/main/asciidoc/deploying-to-kubernetes.adoc

```diff
@@ -860,14 +860,63 @@ implementation("io.quarkus:quarkus-kubernetes-client")
 ----
 
 To access the API server from within a Kubernetes cluster, some RBAC related resources are required (e.g. a ServiceAccount, a RoleBinding).
-So, when the `kubernetes-client` extension is present, the `kubernetes` extension is going to create those resources automatically, so that application will be granted the `view` role.
-If more roles are required, they will have to be added manually.
+So, when the `kubernetes-client` extension is present, by default the `kubernetes` extension is going to create those resources automatically, so that application will be granted the `view` role. You can fully customize the roles and subjects to use using the properties under `quarkus.kubernetes-client.role-binding`.
+If more roles are required, they can create them using the properties under `quarkus.kubernetes.rbac.roles`.
 
 [NOTE]
 ====
 You can disable the RBAC resources generation using the property `quarkus.kubernetes-client.generate-rbac=false`.
 ====
 
+=== Generating RBAC resources
+
+In some scenarios, it's necessary to generate additional https://kubernetes.io/docs/reference/access-authn-authz/rbac/[RBAC] resources. The Kubernetes extension got you covered with the `quarkus.kubernetes.rbac` properties:
+
+[source,properties]
+----
+# Generate Role resource with name "my-role" <1>
+quarkus.kubernetes.rbac.roles.my-role.policy-rules.0.api-groups=extensions,apps
+quarkus.kubernetes.rbac.roles.my-role.policy-rules.0.resources=deployments,deployments/status
```

Comment:
what is the `deployment/status`?

Reply:
I copied it from some Kubernetes official documentation, removing it.

Reply:
Is it a copy paste thing, or it means something I don't know?

Reply:
This is necessary to read the status of deployments. But I copied-pasted from an example from the K8s documentation, we don't need it here.